AEPCore/Sources/configuration/ConfigurationState.swift

```diff
@@ -57,104 +55,81 @@ class ConfigurationState {
         appIdManager = LaunchIDManager(dataStore: dataStore)
     }
 
-    /// Loads the first configuration at launch
-    func loadInitialConfig() {
-        var config = [String: Any]()
-
-        // Load any existing application ID, either saved in persistence or read from the ADBMobileAppID property in the platform's System Info Service.
-        if let appId = appIdManager.loadAppId() {
-            config = configDownloader.loadConfigFromCache(appId: appId, dataStore: dataStore)
-                ?? configDownloader.loadDefaultConfigFromManifest()
-                ?? [:]
-        } else {
-            Log.trace(label: logTag, "App ID not found, attempting to load default config from manifest")
-            config = configDownloader.loadDefaultConfigFromManifest() ?? [:]
+    /// Computes and returns environment aware configuration based on `self.currentConfiguration`
+    /// - Returns: A configuration with the correct values for each key given the build environment, while also
+    ///   removing all keys prefix with `ConfigurationConstants.ENVIRONMENT_PREFIX_DELIMITER`
+    var environmentAwareConfiguration: [String: Any] {
+        queue.sync {
+            return computeEnvironmentConfig()
         }
-
-        updateWith(newConfig: config)
     }
 
-    /// Merges the current configuration to `newConfig` then applies programmatic configuration on top
+    /// Updates the current configuration to `newConfig` then applies programmatic configuration on top
     /// - Parameter newConfig: The new configuration
     func updateWith(newConfig: [String: Any]) {
-        unmergedConfiguration = newConfig
-        currentConfiguration.merge(newConfig) { _, updated in updated }
```

Comment:
why are we changing this logic?

Reply:
None of our APIs support merging the configuration with a provided dictionary. We were only using this functionality within loadInitialConfig(), which simply sets the configuration initially. Even if we had supported this, the current logic would still be incorrect. The unmergedConfiguration should also be merged with the provided config.

Reply:
Ahhhh makes sense. Thanks for the clarification.